flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableFactory.java

```diff
@@ -284,7 +290,9 @@ private void validateConfigOptions(ReadableConfig config) {
 			USERNAME,
 			PASSWORD
 		});
-
+		checkAllOrNone(config, new ConfigOption[]{
```

Comment:
We don't add this check. The `checkAllOrNone` is used for options should be set or not set togehter, e.g. password + username.